core/apiCalls.js

```diff
@@ -176,3 +176,46 @@ export function fetchImageStatusApi(uuid) {
   });
   return p;
 }
+
+export function fetchComposeQueueApi() {
+  const p = new Promise((resolve, reject) => {
+    utils.apiFetch(constants.get_compose_queue)
+    .then(data => {
+      const queue = data.new.concat(data.run);
+      resolve(queue);
```

Comment:
Does this get both composes that are pending plus the one that's in progress?

Reply:
Yes.

Reply:
I've always wondered why you're wrapping those calles in another `Promise`. `utils.apiFetch` already returns a promise, doesn't it?

Reply:
You are correct. I will remove those extra promises from the compose API calls in this PR and will refactor the rest of the API calls in a future fix.